src/include/cache_inline.h

```diff
@@ -511,3 +511,70 @@ __wt_cache_eviction_check(WT_SESSION_IMPL *session, bool busy, bool readonly, bo
 
     return (__wt_cache_eviction_worker(session, busy, readonly, pct_full));
 }
+
+/*
+ * __wt_evict_busy_clr --
+ *     Clear the eviction busy field atomically. Restore the saved cursor if needed.
+ */
+static inline void
+__wt_evict_busy_clr(WT_SESSION_IMPL *session, WT_BTREE *btree, WT_CURSOR *save)
+{
+    (void)__wt_atomic_subv32(&btree->evict_busy, 1);
+    if (save != NULL)
+        /* Restore the caller's history store cursor. */
+        session->hs_cursor = save;
```

Comment:
We probably want to assert `session->hs_cursor == NULL` here.  If the caller used a history store cursor, they should have closed it (which will clear that field).  If not, this assignment will leak the cursor they created.